CRM/Admin/Form/CMSUser.php

```diff
@@ -25,6 +25,16 @@ class CRM_Admin_Form_CMSUser extends CRM_Core_Form {
    */
   public $submitOnce = TRUE;
 
+  /**
+   * Disable on Standalone
+   */
+  public function preProcess() {
+    $config = CRM_Core_Config::singleton();
+    if ($config->userFramework === 'Standalone') {
+      throw new \Exception('No CMS to sync users to when using CiviCRM Standalone');
+    }
```

Comment:
I know this is a draft but just noting two things:
* A year or two ago we tried to remove most of the `if Drupal` from various files to make it more OO-ey. I know in standalone sometimes it hasn't booted yet and so there's no choice, but here I think it would be better to do something like `if !$config->userSystem->allowsUserSync()` and then CRM_Utils_System_Base::allowsUserSync would return true, and CRM_Utils_System_Standalone::allowsUserSync() would return false. In theory this also allows mocking both variations in backend unit tests (although it's not currently set up for more than one unittest mock).
* In forms we usually use CRM_Core_Error::statusBounce instead of throwing exceptions. Even if it were to throw an exception it's preferred to use CRM_Core_Exception.

Reply:
thanks for these pointers :+1:

Reply:
This got me thinking -- it's specifically standaloneusers that the syncing is incompatible with. So I've made it conditional on that, leaving open the door for alternatives where syncing is a thing (external auth providers maybe?)